coco/dragon: add midi cartridge

tlindner opened the pull request:
Implement a midi device as a coco/dragon cartridge.

rb6502 commented:
This looks good as far as 6850 MIDI goes.  Have you tested it yourself?

cuavas commented on src/devices/bus/coco/coco_midi.cpp:
```diff
@@ -0,0 +1,130 @@
+    m_acia->reset();
+}
+
+WRITE_LINE_MEMBER( coco_dragon_midi_device::write_acia_clock )
+{
+    m_acia->write_txc(state);
+    m_acia->write_rxc(state);
+}
+
+WRITE_LINE_MEMBER(coco_dragon_midi_device::midi_in)
+{
+    // MIDI in signals is sent to both the 6850 and the MIDI thru output port
+    m_acia->write_rxd(state);
+    m_mdthru->write_txd(state);
+}
```
You can do all this in machine config, there’s no need for additional functions.

cuavas commented on src/devices/bus/coco/coco_midi.cpp:
```diff
@@ -0,0 +1,130 @@
+#include "machine/clock.h"
+#include "bus/midi/midi.h"
+
+#define MC6850_TAG "mc6850"
```
It’s pointless having this macro when you use the finder to access it everywhere anyway.  The literal tag wouldn’t be repeated even without the macro.

cuavas commented on src/devices/bus/coco/coco_midi.cpp:
```diff
@@ -0,0 +1,130 @@
+            write8sm_delegate(m_acia, FUNC(acia6850_device::write)));
+}
+
+void coco_dragon_midi_device::device_reset()
+{
+    m_acia->reset();
+}
```
Reset propagates to child devices automatically.  You shouldn’t need to be doing this.

cuavas reviewed (changes requested):
I’ve said it before, I’ll say it again – if a class is declared inside an anonymous namespace, it should be implemented inside the anonymous namespace.  Also, namespaces are not declarations and hence don’t need a terminating semicolon.

tlindner commented:
@rb6502 I've tested midi out. I really should test midi in before this is committed. I will report back.
@cuavas I believe I've addresses all your comments.

cuavas commented on src/devices/bus/coco/coco_midi.cpp:
```diff
@@ -0,0 +1,114 @@
+				write8sm_delegate(m_acia, FUNC(acia6850_device::write)));
+	}
+
+	WRITE_LINE_MEMBER( coco_dragon_midi_device::write_acia_clock )
+	{
+		m_acia->write_txc(state);
+		m_acia->write_rxc(state);
+	}
```
You can do this in machine config as well.

tlindner commented:
I see that I am not validating. Looking into it now.
